### PR TITLE
Modified AssignmentHandler to return false for Tablets with OpId

### LINKED ISSUE #3741: Avoid serving tablets with an operation id in the scan server
Tablets that are splitting ( and in the future merging or deleting) will have an operation id set.  When this is set it means a destructive operations is active on the tablet.  If a scan server reads a tablets metadata and the operation id is set, then it should probably not serve queries from the files in that tablet.  This is because those files may be in flux and not represent all of the tablets data while the operation is in progress which may modify the tablet and its files over multiple mutations.  Its ok for scan servers to respond with old data, but they should not knowingly respond with data that may be incomplete.

### PR DESCRIPTION
**Pull request by dlmarion**
In #3425 the Manager was modified to set the TabletGoalState to UNASSIGNED for Tablets that have an OperationId. This changes the AssignmentHandler to return false in the same condition. This will cause both the TabletServer and the ScanServer to not host a Tablet when there is an OperationId.

Closes #3741

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java [hunk 1]
```diff
@@ -285,6 +285,11 @@ public static boolean checkTabletMetadata(KeyExtent extent, TServerInstance inst
       return false;
     }
 
+    if (meta.getOperationId() != null) {
+      log.info(METADATA_ISSUE + "metadata entry has a FATE operation id");
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java:289]: Can add some info to the logging

```suggestion
      log.info(METADATA_ISSUE + "metadata entry has a FATE operation id {} {} {}", extent, loc, meta.getOperationId());
```

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java [hunk 2]
```diff
@@ -246,6 +255,51 @@ public void testScanWithTabletHostingMix() throws Exception {
...
+              Iterables.size(scanner);
+            }));
+
+        // Confirm that the TabletServer will not complete the scan
+        scanner.setConsistencyLevel(ConsistencyLevel.IMMEDIATE);
+        assertThrows(AssertionFailedError.class,
+            () -> assertTimeoutPreemptively(Duration.ofSeconds(30), () -> {
+              Iterables.size(scanner);
+            }));
+      } // when the scanner is closed, all open sessions should be closed
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:299]: Seems like this code will wait 30 seconds and then wait 30 seconds.  Maybe could refactor the code to do something like the following.

 1. create executor w/ two threads
 2. future1 = executor.submit( eventual scan )
 3. future2 = executor.submit( immediate scan )
 4. wait 30 secs
 5. assert futures are not done and cancel them

This allows the scans to run concurrently and only have a single wait.

**Review by keith-turner — APPROVED** [review]
Its nice the test exercises immediate and eventual scan.

File: test/src/main/java/org/apache/accumulo/test/ScanServerIT.java (hunk 2 above)
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/ScanServerIT.java:299]: Could take things a bit further and combine the batch scan and regular scan with something like the following

 1. create executor w/ four threads
 2. future1 = executor.submit( eventual scan )
 3. future2 = executor.submit( immediate scan )
 4. future3 = executor.submit( eventual batch scan )
 5. future4 = executor.submit( immediate batch scan )
 6. wait 30 secs
 7. assert futures are not done and cancel them

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/ScanServerIT.java]: Implemented in 019cd50

File: server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java]: Implemented in 019cd50

**Pull request merged**